app/http.php

```diff
@@ -0,0 +1,346 @@
+        }
+
+        $hostname = $option->getState('hostname') ?? '';
+
+        $client = new Client($hostname, 80);
+        $client->setMethod($request->getMethod());
+
+        $headers = \array_merge($request->getHeaders(), [
+            'authorization' => 'Bearer ' . App::getEnv('OPEN_RUNTIMES_PROXY_EXECUTOR_SECRET', '')
+        ]);
+
+        // Header used for testing
+        if (App::isDevelopment()) {
+            $headers = \array_merge($headers, [
+                'x-open-runtimes-executor-hostname' => $hostname
+            ]);
+        }
+
+        $client->setHeaders($headers);
+        $client->setData($request->getRawPayload());
+        $client->execute($request->getURI());
+
+        foreach ($client->headers as $header => $value) {
+            $response->addHeader($header, $value);
+        }
+
+        $response
+            ->setStatusCode(\intval($client->getStatusCode()))
+            ->send(\strval($client->getBody()));
+    });
+
+App::error()
+    ->inject('utopia')
+    ->inject('error')
+    ->inject('logger')
+    ->inject('request')
+    ->inject('response')
+    ->action(function (App $utopia, throwable $error, ?Logger $logger, Request $request, Response $response) {
+        $route = $utopia->match($request);
+        logError($error, "httpError", $logger, $route);
+
+        switch ($error->getCode()) {
+            case 400: // Error allowed publicly
+            case 401: // Error allowed publicly
+            case 402: // Error allowed publicly
+            case 403: // Error allowed publicly
+            case 404: // Error allowed publicly
+            case 406: // Error allowed publicly
+            case 409: // Error allowed publicly
+            case 412: // Error allowed publicly
+            case 425: // Error allowed publicly
+            case 429: // Error allowed publicly
+            case 501: // Error allowed publicly
+            case 503: // Error allowed publicly
+                $code = $error->getCode();
+                break;
+            default:
+                $code = 500; // All other errors get the generic 500 server error status code
+        }
+
+        $output = [
+            'message' => $error->getMessage(),
+            'code' => $error->getCode(),
+            'file' => $error->getFile(),
+            'line' => $error->getLine(),
+            'trace' => $error->getTrace(),
+            'version' => App::getEnv('OPEN_RUNTIMES_PROXY_VERSION', 'UNKNOWN')
+        ];
+
+        $response
+            ->addHeader('Cache-Control', 'no-cache, no-store, must-revalidate')
+            ->addHeader('Expires', '0')
+            ->addHeader('Pragma', 'no-cache')
+            ->setStatusCode(\intval($code));
+
+        $response->json($output);
+    });
+
+/** @phpstan-ignore-next-line */
+Co\run(
+    function () use ($register) {
+        /** @phpstan-ignore-next-line */
+        \Swoole\Coroutine\batch([
+                function () use ($register) {
```

Comment:
I don't think this is the right way to use go functions or batch for that matter

Co\run {
  batch {
      go 
   }
}

I dont think this code is really doing what we intend it to do. 

Batch takes a list of functions whereas here its only taking a **list of 1 function**. I dont think its making sense.

Reply:
Replaced batch with barrier. Removed go method.